internal/monitors/elasticsearch/client/indexstatsfetcher.go

```diff
@@ -87,54 +87,56 @@ func getIndexStatsHelper(indexStats IndexStats, defaultDims map[string]string, e
+		out = append(out, indexStatsGroup.Store.getIndexStoreStats(enhancedStatsForIndexGroupsOption[StoreStatsGroup], defaultDimsForPrimaries)...)
+		out = append(out, indexStatsGroup.Indexing.getIndexIndexingStats(enhancedStatsForIndexGroupsOption[IndexingStatsGroup], defaultDimsForPrimaries)...)
+		out = append(out, indexStatsGroup.Get.getIndexGetStats(enhancedStatsForIndexGroupsOption[GetStatsGroup], defaultDimsForPrimaries)...)
+		out = append(out, indexStatsGroup.Search.getIndexSearchStats(enhancedStatsForIndexGroupsOption[SearchStatsGroup], defaultDimsForPrimaries)...)
+		out = append(out, indexStatsGroup.Merges.getIndexMergesStats(enhancedStatsForIndexGroupsOption[MergesStatsGroup], defaultDimsForPrimaries)...)
+		out = append(out, indexStatsGroup.Refresh.getIndexRefreshStats(enhancedStatsForIndexGroupsOption[RefreshStatsGroup], defaultDimsForPrimaries)...)
+		out = append(out, indexStatsGroup.Flush.getIndexFlushStats(enhancedStatsForIndexGroupsOption[FlushStatsGroup], defaultDimsForPrimaries)...)
+		out = append(out, indexStatsGroup.Warmer.getIndexWarmerStats(enhancedStatsForIndexGroupsOption[WarmerStatsGroup], defaultDimsForPrimaries)...)
+		out = append(out, indexStatsGroup.QueryCache.getIndexQueryCacheStats(enhancedStatsForIndexGroupsOption[QueryCacheStatsGroup], defaultDimsForPrimaries)...)
+		out = append(out, indexStatsGroup.FilterCache.getIndexFilterCacheStats(enhancedStatsForIndexGroupsOption[FilterCacheStatsGroup], defaultDimsForPrimaries)...)
+		out = append(out, indexStatsGroup.Fielddata.getIndexFielddataStats(enhancedStatsForIndexGroupsOption[FieldDataStatsGroup], defaultDimsForPrimaries)...)
+		out = append(out, indexStatsGroup.Completion.getIndexCompletionStats(enhancedStatsForIndexGroupsOption[CompletionStatsGroup], defaultDimsForPrimaries)...)
+		out = append(out, indexStatsGroup.Segments.getIndexSegmentsStats(enhancedStatsForIndexGroupsOption[SegmentsStatsGroup], defaultDimsForPrimaries)...)
+		out = append(out, indexStatsGroup.Translog.getIndexTranslogStats(enhancedStatsForIndexGroupsOption[TranslogStatsGroup], defaultDimsForPrimaries)...)
+		out = append(out, indexStatsGroup.RequestCache.getIndexRequestCacheStats(enhancedStatsForIndexGroupsOption[RequestCacheStatsGroup], defaultDimsForPrimaries)...)
+		out = append(out, indexStatsGroup.Recovery.getIndexRecoveryStats(enhancedStatsForIndexGroupsOption[RecoveryStatsGroup], defaultDimsForPrimaries)...)
+		out = append(out, indexStatsGroup.IDCache.getIndexIDCacheStats(enhancedStatsForIndexGroupsOption[IDCacheStatsGroup], defaultDimsForPrimaries)...)
+		out = append(out, indexStatsGroup.Suggest.getIndexSuggestStats(enhancedStatsForIndexGroupsOption[SuggestStatsGroup], defaultDimsForPrimaries)...)
+		out = append(out, indexStatsGroup.Percolate.getIndexPercolateStats(enhancedStatsForIndexGroupsOption[PercolateStatsGroup], defaultDimsForPrimaries)...)
 	}
 
+	indexStatsGroup := indexStats.Total
```

Comment:
You can just assign `indexStatsGroup` to either the total or primaries struct in a conditional the top of this method , along with the dimensions, and then avoid all the duplication of the `out = append(out ...` lines.

Reply:
Moved the block to a separate method.